### DISCUSSION TITLE
[Feature] Publish service plugin for Lightroom Classic

### DISCUSSION THREAD

**Discussion by shumafuk** [immich-app/immich#8164] [Feature Request]
### The feature

I find maintaining database of imported pictures quite challenging in my workflow. Currently, when I need to update image on server, I need to find it, delete it, find it in my export folder (synced with Lightroom Classic), re-upload it and put it in album again. Also folder structure that I use for my picture database (by folders) needs to be recreated in Immich manually.

To solve this complicated workflow, publish service plugin for Lightroom Classic would solve many issues with keeping Immich database updated.

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by etnoy** [comment]
If anyone wants to create a Lightroom plugin, that would be great. I don't think that's something the core team is focusing on, though.

**Reply by justkowal** [reply]
As I suggested the feature request I may clarify, this request should be more focused on providing way of storing edited and original copies of an image. In the future, built-in editor could tap into this mechanism and for now API and a button on mobile apps that would enable editing in external apps like PicsArt could be added.

To summarize:
- Add an option that will allow storing edited/original version of an image. In all standard contexts, the main image is the edited one 
- Add an option to get the original, unedited file both via API and the UI
- Add a button in the mobile app that will allow to open external editor like PicsArt and then save the edit as an edited version on the immich server.

Why not use bundles?
Bundles are already a multi-purpose option and if at any time immich would get a built-in editor, the feature to elegantly handle original, unedited files will be already present.

**Comment by bmachek** [comment]
https://github.com/bmachek/lrc-immich-plugin
